sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -77,7 +77,7 @@ abstract class FileScanBuilder(
     this.dataFilters = dataFilters
     val translatedFilters = mutable.ArrayBuffer.empty[sources.Filter]
     for (filterExpr <- dataFilters) {
-      val translated = DataSourceStrategy.translateFilter(filterExpr, true)
+      val translated = DataSourceStrategy.translateFilter(filterExpr, true, true)
```

Comment:
Before:
```
== Physical Plan ==
*(1) Project [NAME#7, THEID#8]
+- *(1) Filter (((THEID#8 > 0) AND (trim(NAME#7, None) = mary)) OR (THEID#8 > 10))
   +- BatchScan json file:/private/var/folders/j0/b7tktmwj1pl591yd6ysvv5sm0000gq/T/spark-ea31f369-4c0e-442b-941d-bbde3be2f41b[NAME#7, THEID#8] JsonScan DataFilters: [(((THEID#8 > 0) AND (trim(NAME#7, None) = mary)) OR (THEID#8 > 10))], Format: json, Location: InMemoryFileIndex(1 paths)[file:/private/var/folders/j0/b7tktmwj1pl591yd6ysvv5sm0000gq/T/spark-ea..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<NAME:string,THEID:int> RuntimeFilters: []

```
After:
```
== Physical Plan ==
*(1) Project [NAME#7, THEID#8]
+- *(1) Filter (((THEID#8 > 0) AND (trim(NAME#7, None) = mary)) OR (THEID#8 > 10))
   +- BatchScan json file:/private/var/folders/j0/b7tktmwj1pl591yd6ysvv5sm0000gq/T/spark-aaa18d17-865a-45d0-9bc1-e8e90855add5[NAME#7, THEID#8] JsonScan DataFilters: [(((THEID#8 > 0) AND (trim(NAME#7, None) = mary)) OR (THEID#8 > 10))], Format: json, Location: InMemoryFileIndex(1 paths)[file:/private/var/folders/j0/b7tktmwj1pl591yd6ysvv5sm0000gq/T/spark-aa..., PartitionFilters: [], PushedFilters: [Or(GreaterThan(THEID,0),GreaterThan(THEID,10))], ReadSchema: struct<NAME:string,THEID:int> RuntimeFilters: []

```

Reply:
`PushedFilters` is not empty:
```
PushedFilters: [Or(GreaterThan(THEID,0),GreaterThan(THEID,10))]
```

Reply:
I feel we can partially push down for each data sources.(contains JDBC database).
Please tell me the exceptions, if not possible.

Reply:
These are two things.